### PR TITLE
Attach changelog to release automatically

### LINKED ISSUE #4636: Release process: automatically add changelog to the GitHub release
This issue is about updating the release process so the changelog is automatically added to the GitHub release description. Notice that the changelog is already automatically generated and the GitHub release is also created automatically by the release script, so the only missing piece is adding that changelog to the release description.

### PR DESCRIPTION
**Pull request by gigitux**
This PR fixes woocommerce/woocommerce-gutenberg-products-block#4636

## Manual test

1. Create a new repo. You can start from [this template](https://github.com/gigitux/woo-test-repo).
2. Set a workflow for `woocoomerce-automation`. You can copy-paste [this file](https://github.com/gigitux/woocommerce-automation-test-repo/blob/main/.github/workflows/automation.yml). The important thing is to change the value `uses`. It has to point to this branch.
3. Create a new milestone and give the title a number version (e.g.: 1.0).
4. Create a PR with a changelog, give the name `Release: 1.0`([example](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6129)). It would be great if you copy the entire PR body of the linked PR.
5. Create a release with Github UI, and give as the release title the version (for example 1.0).
6. Now, check that the action updates the release description and put the changelog.

### CODE REVIEW

File: lib/automations/release/branch-create-handler.js [hunk 1]
```diff
@@ -116,6 +117,7 @@ const branchHandler = async ( context, octokit, config ) => {
 		);
 		changelog = await getChangelog( changelogItems, config );
 		devNoteItems = getDevNoteItems( changelogItems, config );
+		await attachChangelogToRelease( context, octokit, changelog );
```
Comment (nerrad) [inline lib/automations/release/branch-create-handler.js]: The GitHub release won't have been created yet by this point right? Won't this just attach this to the previous release?

**Review by nerrad — CHANGES_REQUESTED** [review]
I think there may be a misunderstanding of what goal we're shooting for here.

The release automation currently in this project is mostly just creating an opinionated "release Pull Request" triggered by the creation of a branch for handling a release build. At the point of this release PR being created, there won't necessarily be any corresponding GitHub release branch for the PR (in the case of WooCommerce Blocks, the GH release is triggered by a script executed by whoever is doing the release, later in the process).

So as is, this PR doesn't solve the issue because it will end up adding the changelog to the _previous_ release (not the one being worked on).

One thing you could investigate is whether you could have an automation triggered on the [`releaseevent`](https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types#releaseevent) - hopefully you could get enough information from that payload to generate and attach the changelog?

**Review by alexflorisca — APPROVED** [review]
Good work @gigitux. The only thing I'd say is that this action looks for a specific format in our Woo Blocks Release PRs, which might not apply to Woo core or other Woo products. I'm happy for us to leave it as is though because other teams can choose to use it or not. If we give it a more specific name, then other teams start using it, it won't be very clear. 

Merge it and let's try it out on the next Woo Blocks release 👍

**Comment by gigitux** [comment]
> The only thing I'd say is that this action looks for a specific format in our Woo Blocks Release PRs, which might not apply to Woo core or other Woo products.

The specific format that we use in our Woo Blocks Release PRs is generated automatically by this automation repo (from the creation of the PR to the creation of the release). I guess that the format is always the same if a team adopts this automation.

**Pull request merged**